tests/integration/targets/iso_customize/tasks/iso_customize.yml

```diff
@@ -0,0 +1,85 @@
+# Copyright (c) 2022, Ansible Project
+# Copyright (c) 2022, VMware, Inc. All Rights Reserved.
+# GNU General Public License v3.0+ (see LICENSES/GPL-3.0-or-later.txt or https://www.gnu.org/licenses/gpl-3.0.txt)
+# SPDX-License-Identifier: GPL-3.0-or-later
+
+- name: "Customize ISO file"
+  community.general.iso_customize:
+    src_iso: "{{ test_dir }}/test.iso"
+    dest_iso: "{{ test_dir }}/{{ dest_iso_name }}"
+    delete_files:
+      - "/test01.cfg"
+    add_files:
+      - src_file: "{{ test_dir }}/test01.cfg"
+        dest_file: "/preseed/ubuntu.seed"
+  register: customize_iso_result
+
+- name: "Display the ISO customized result"
+  debug: var=customize_iso_result
+
+- name: Test customize iso result
+  assert:
+    that:
+    - customize_iso_result is not failed
+
+- debug: var=ansible_distribution
+
+# Only need to check the files is deleted/added as the requested modifications 
+# in one type of OS(take MacOS here)
+- block:
+    - name: Create sub directory to mount customized ISO
+      file:
+        path: '{{ test_dir }}/iso_mount'
+        state: directory
+
+    - name: "Mount customized ISO on MAC"
+      shell: "hdiutil attach {{ test_dir }}/{{ dest_iso_name }} -mountroot {{ test_dir }}/iso_mount"
+
+    # For MAC, we have different root directory for different type of ISO
+    - set_fact:
+        mount_root_dir: "{{ test_dir }}/iso_mount/disk_image"
+
+    - set_fact:
+        mount_root_dir: "{{ test_dir }}/iso_mount/AUTOINSTALL"
+      when: dest_iso_name.find('joliet') != -1
+
+    - set_fact:
+        mount_root_dir: "{{ test_dir }}/iso_mount/CDROM"
+      when: dest_iso_name.find('udf') != -1
+
+    - debug: var=mount_root_dir
+
+    - name: Check the file test01.cfg is deleted
+      stat:
+        path: "{{ mount_root_dir }}/test01.cfg"
+      register: check_file
+
+    - assert:
+        that:
+        - check_file.stat.exists == False
+
+    - name: Check the file /preseed/ubuntu.seed is added
+      stat:
+        path: "{{ mount_root_dir }}/preseed/ubuntu.seed"
+      register: check_file
+
+    - assert:
+        that:
+        - check_file.stat.exists == True
+
+    - name: Umount ISO
+      mount:
+        path: "{{ mount_root_dir }}"
+        fstab: "{{ test_dir }}/temp.fstab"
+        state: unmounted
+
+    - name: Delete directory
+      file:
+        path: "{{ test_dir }}/iso_mount"
+        state: absent
+  when: ansible_distribution == "MacOSX"
+
+- name: Delete iso file after testing
+  file:
+    path: "{{ test_dir }}/test.iso"
+    state: absent
```

Comment:
Readability would be improved by moving these "tear down" actions to a separate file - similarly to the `prepare.yml` but to do clean-up tasks.

Reply:
Thanks very much. I tried and find there is no necessary for a separate file  "cleanup.yml".

1) After testing every type of ISO, we need to umount it.
2) After testing every type of ISO, there is no need to delete sub dir "iso_mount" anymore
3) After testing every type of ISO, we don't need to delete test.iso because the module "iso_create" will overwrite the file test.iso if it exists.